RocketChat/Rocket.Chat
# fix: SAML "Overwrite user fullname" setting is not being honored

## Proposed changes (including videos or screenshots)

 - Check for SAML "Overwrite user fullname" setting before updating a user identity coming from a SAML login.
 - Add end-to-end tests to validate that names are only overwritten when the setting is enabled (and also to make sure they won't be updated along with the username when the latter is changed)

## Issue(s)


## Steps to test or reproduce

 1. Create a user in RC;
 2. Login using SAML with a user that has the same name or username (following the **"Immutable field name"** setting in **SAML > General > Behavior** ) as the previously created user, but a distinct name;

 **_Expected behavior:_** the user's full name should only be updated/overwritten in case the **"Overwrite user fullname (use idp attribute)"** setting is enabled. Otherwise, it shouldn't be changed (even if the username is updated on login).
**_Current behavior_** the user's full name is updated regardless of the **"Overwrite user fullname (use idp attribute)"** setting.

## Further comments

[SUP-605](https://rocketchat.atlassian.net/browse/SUP-605)

[SUP-605]: https://rocketchat.atlassian.net/browse/SUP-605?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
